lib/modules/datasource/pypi/index.ts

```diff
@@ -34,45 +34,51 @@ export class PypiDatasource extends Datasource {
     packageName,
     registryUrl,
   }: GetReleasesConfig): Promise<ReleaseResult | null> {
-    let dependency: ReleaseResult | null = null;
+    const dependency: ReleaseResult = { releases: [] };
     // TODO: null check (#22198)
-    const hostUrl = ensureTrailingSlash(
-      registryUrl!.replace('https://pypi.org/simple', 'https://pypi.org/pypi'),
-    );
+    let hostUrl = ensureTrailingSlash(registryUrl!);
     const normalizedLookupName = PypiDatasource.normalizeName(packageName);
 
-    // not all simple indexes use this identifier, but most do
-    if (hostUrl.endsWith('/simple/') || hostUrl.endsWith('/+simple/')) {
+    // convert pypi json api url to simple
+    hostUrl = hostUrl.replace(
+      'https://pypi.org/pypi',
+      'https://pypi.org/simple',
+    );
+
+    const simpleFound = await this.addResultsViaSimple(
+      normalizedLookupName,
+      hostUrl,
+      dependency,
+    ).catch((err) => {
+      if (err.statusCode !== 404) {
+        throw err;
+      }
       logger.trace(
-        { packageName, hostUrl },
-        'Looking up pypi simple dependency',
+        'Simple api not found. Looking up pypijson api as fallback.',
```

Comment:
Why do you have removed the variables here?

Reply:
I forgot those to include in the logging. Now fixed